api/server.py

```diff
@@ -324,40 +331,50 @@ def get_events_data(event_type):
 
     module_name = get_value_from_request("module_name")
     date = get_value_from_request("date")
-    filter = get_value_from_request("filter")
+    filter_by = get_value_from_request("filter")
 
     try:
-        query = filter_by_date(date) if date else {}
-        query.update(filter_by_module_name(module_name) if module_name else {})
-        query.update(
-            {
-                key: filter_by_regex(fix_filter_query(filter)[key]) for key in fix_filter_query(filter)
-            } if filter else {}
-        )
-
-        return jsonify(
-            {
-                "total": event_types[event_type].count(query),
-                "data": [
-                    i for i in
-                    event_types[event_type].find(
-                        query,
-                        {
-                            "_id": 0
-                        }
-                    ).skip(
-                        fix_skip(
-                            get_value_from_request("skip")
-                        )
-                    ).limit(
-                        fix_limit(
-                            get_value_from_request("limit")
-                        )
-                    )
-                ]
+        query = {
+            "query": {
+                "bool": {
+                    "must": [
+                        filter_by_date(date)['query']
+                    ] if date else [],
+                    "filter": []
+                }
             }
-        ), 200
-    except Exception:
+        }
+        if module_name:
+            query['query']['bool']['must'].append(
```

Comment:
can you please explain where's the must coming from or what's the must thing?

Reply:
must | The clause (query) must appear in matching documents and will contribute to the score.
-- | --

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-bool-query.html

When we need to query with multiple fields we use bool query in elasticsearch.